azurelinuxagent/distro/ubuntu/loader.py

```diff
@@ -22,11 +22,25 @@
 def get_osutil():
     from  azurelinuxagent.distro.ubuntu.osutil import Ubuntu1204OSUtil, \
                                                       UbuntuOSUtil, \
-                                                      Ubuntu14xOSUtil
+                                                      Ubuntu14xOSUtil, \
+                                                      UbuntuSnappyOSUtil
+    import os
+    import azurelinuxagent.utils.fileutil as fileutil
+
+    # TODO: https://bugs.launchpad.net/snappy/+bug/1481086
+    def is_snappy():
```

Comment:
Please check azurelinuxagent.metadata. You could use DISTRO_FULL_NAME to detect snappy.